src/Auditor.php

```diff
@@ -79,10 +81,49 @@ public function execute(AuditableContract $model)
             return;
         }
 
-        if ($audit = $driver->audit($model)) {
+        $uuid = null;
+        $list_of_properties = [];
+        foreach (config('audit.relation_hierarchy', []) as $class_type_to_audit => $local_list_of_methods) {
+            foreach ($local_list_of_methods as $local_method) {
+                if ($model instanceof $class_type_to_audit) {
+                    $list_of_properties[] = $local_method['property'];
+                    $uuid = (string) Uuid::generate();
+                } elseif ($model instanceof $class_type_to_audit) {
+                    if ($model instanceof $local_method['yields']) {
+                        $list_of_properties[] = $local_method['relator'];
+                        $uuid = (string) Uuid::generate();
+                    }
+                }
+            }
+        }
+
+        if ($audit = $driver->audit($model, $uuid, false)) {
             $driver->prune($model);
         }
 
+        if ($uuid) {
```

Comment:
Nope.

Reply:
Please elaborate? I'm using is because I need to logically group the audit and 'related' audit records generated by a particular change to a model.